tournamentcontrol/competition/tests/test_logo_fields.py

```diff
@@ -0,0 +1,273 @@
+
+    def test_season_logo_via_admin(self):
+        """Test that Season edit view includes logo fields."""
+        namespace = self.season._get_admin_namespace()
+        args = self.season._get_url_args()
+        
+        with self.login(self.superuser):
+            response = self.get(f"{namespace}:edit", *args)
+            self.response_200()
+            self.assertResponseContains('name="logo_colour"', html=False)
+            self.assertResponseContains('name="logo_monochrome"', html=False)
+
+    def test_division_logo_via_admin(self):
+        """Test that Division edit view includes logo fields."""
+        namespace = self.division._get_admin_namespace()
+        args = self.division._get_url_args()
+        
+        with self.login(self.superuser):
+            response = self.get(f"{namespace}:edit", *args)
+            self.response_200()
+            self.assertResponseContains('name="logo_colour"', html=False)
+            self.assertResponseContains('name="logo_monochrome"', html=False)
+
+    def test_team_logo_via_admin(self):
+        """Test that Team edit view includes logo fields."""
+        namespace = self.team._get_admin_namespace()
+        args = self.team._get_url_args()
+        
+        with self.login(self.superuser):
+            response = self.get(f"{namespace}:edit", *args)
+            self.response_200()
+            self.assertResponseContains('name="logo_colour"', html=False)
+            self.assertResponseContains('name="logo_monochrome"', html=False)
+
+    @override_settings(DEFAULT_FILE_STORAGE="django.core.files.storage.InMemoryStorage")
+    def test_competition_logo_upload_png(self):
+        """Test that Competition accepts PNG file uploads via admin POST."""
+        from django.core.files.storage import default_storage
+        
+        png_file = SimpleUploadedFile(
+            "competition_logo.png", self.png_data, content_type="image/png"
+        )
+        
+        # Use assertGoodEditView which handles form submission properly
+        self.assertGoodEditView(
+            f"{self.competition._get_admin_namespace()}:edit",
+            *self.competition._get_url_args(),
+            data={
+                "title": self.competition.title,
+                "slug": self.competition.slug,
+                "logo_colour": png_file,
+            }
+        )
+        
+        # Verify file was stored
+        self.competition.refresh_from_db()
+        self.assertTrue(self.competition.logo_colour.name)
+        self.assertIn("logos/competition/colour/", self.competition.logo_colour.name)
+        self.assertIn("competition_logo", self.competition.logo_colour.name)
+        
+        # Verify file exists and contents match
+        self.assertTrue(default_storage.exists(self.competition.logo_colour.name))
+        stored_content = default_storage.open(self.competition.logo_colour.name).read()
+        self.assertEqual(stored_content, self.png_data)
```

Comment:
Use the File API instead of the Storage API.

This is a FileField, so can call exists and open directly on the field.